cfpb/owning-a-home
# JS build process

This is a potential approach to simplify the JS build process by automatically adding shared javascript (for now, the nemo code) to the base template, and optionally adding page-specific js via a new `page_js` variable.

### Changes
- remove the `entries` entry from `factor-bundle` -- this seems to be unnecessary?
- add `base.js` file that requires the nemo code
- remove nemo requirements from all page-specific js, and remove the page-specific files that were only adding nemo
- conditionally add page-specific js in base template
- use different variable for page-specific js to avoid potential conflict with sidebar `active-page` usage

### Testing
- Check that page-specific js is still loaded on `/check-rates`, `/loan-comparison`, `/loan-options`, and `/process` (did I miss anything?)
- Check that the mobile menu still works on all OAH pages

### Review
@cfarm @amymok 

### Notes
- I'm submitting this to start discussion; it doesn't need to be merged right away.
